core/src/main/scala/org/bykn/bosatsu/SourceConverter.scala

```diff
@@ -1218,7 +1218,7 @@ final class SourceConverter(
 
   private def anonNameStrings(): Iterator[String] =
     rankn.Type.allBinders.iterator
-      .map(_.name)
+      .map { bn => "_" + bn.name }
```

Comment:
this is a weird hidden hack: we are making names that are not parseable to make synthetic names.

Probably a better solution is to just make a new Bindable type called Synthetic, that has no Parser, so it is never the result of a parsing rather than backdoor it here.